README.md

```diff
@@ -0,0 +1,57 @@
+# Coveo Cloud services' official `.js` client
+
+The main goal of this package is to provide an easy to configure and straightforward way of querying any API fronted by the Coveo Cloud's services using javascript.
+
+## Getting started
+
+### Install
+
+```bash
+npm install @coveord/platform-client
+# Typescript types are included in the package
+```
+
+### Configure
+
+```js
+const platform = new CoveoPlatform({
+    /* configuration options */
+});
+```
+
+### Use
+
+```js
+platform.resource.action({
+    /* action dependant options */
+});
+```
+
+Every action returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) object.
+
+### Example
+
+```js
+const platform = new CoveoPlatform({
+    organizationId: 'some-coveo-platform-organization-id',
+    accessTokenRetreiver: () => 'your-coveo-platform-access-token-or-api-key',
+});
+
+// listing the organization's catalogs
+platform.catalog.list({page: 0, pageSize: 10}).then((res) => {
+    console.log(JSON.stringify(res));
+});
+```
+
+## Documentation
+
+The whole project is built using Typescript and exposes every type needed so that you typically won't need to refer to any documentation here. API discovery throught code is a very nifty feature so we strongly recommand using this package in a Typescript environment. For that specific reason, the decision has been made not to document any option, resource or action except the main configuration options. For in-depth explanations about each and every API exposed by this package, please consult our [official documentation portal](https://docs.coveo.com/en/151/cloud-v2-developers/coveo-cloud-v2-for-developers).
```

Comment:
What do you mean by `exposes every type` and `API discovery through code`?

Reply:
`exposes every type`: Typescript generates type declarations in a separate file which is exposed by our configuration, therefore usable by anyone that imports our package in his code base.

`API discovery through code`: IDEs typically propose some auto completions based on Typescript type declarations which enables to "discover" all the API supported in our package. This means a user of our package doesn't need to know what API exists or not, he can discover them without leaving his IDE.

Reply:
Thanks! Then let me suggest:
This project is built using TypeScript and automatically generates relevant type declarations. Most TypeScript IDEs will display those type declarations as autocompletions, so that you typically will not need to refer to external documentation. Hence, the decision has been made not to document any option, resource, or action, except for the main configuration options. For in-depth documentation on the APIs exposed by this package, please consult our [official documentation portal](https://docs.coveo.com/en/151/cloud-v2-developers/coveo-cloud-v2-for-developers).